### ISSUE TITLE
Read requests on the ledger API succeed for unallocated parties

### ISSUE THREAD

**Issue by cocreature** [digital-asset/daml#8461]
Read requests (for example on the ACS service or transaction service) succeed even if the party you are reading as has not been allocated. This seems like a bug to me:

Steps to reproduce:
1. `daml sandbox --implicit-party-allocation=false --ledgerid l`
2. `grpcurl -d '{"ledger_id": "l", "filter": {"filtersByParty": {"Alice": {"inclusive": {}}}}}' -plaintext localhost:6865 com.daml.ledger.api.v1.ActiveContractsService/GetActiveContracts`

Expected behavior:
Request gets rejected because `Alice` has not been allocated.

Actual behavior:
Empty result:
```
{
  "offset": "00000000000000010000000000000000"
}
```

**Comment by rautenrieth-da** [comment]
When implementing multi-party submissions, we discussed having the submission service validate that all `readAs` and `actAs` parties are allocated on the participant. We decided not to because:

1. It incurs some runtime cost (extra database lookups for validating submissions)
2. We assume that the ledger operator also controls the access token provider, and the token provider should not hand out tokens for parties that are not allocated on the participant
3. It does not affect the DAML ledger model, as the ledger backend already validates that all `actAs` parties are allocated on the submitting participant, and the `readAs` parties are not part of the DAML ledger model.

A similar argument could be made for validating parties in ACS/TX requests. I don't think an empty result is strictly wrong, but I agree that an explicit error would be more user friendly.

**Comment by cocreature** [comment]
Yeah the token argument is fair. It still seems a bit fishy to me that we rely on party allocations and tokens being in sync while not enforcing it but if the performance costs are too high, it might just be something we have to live with.

**Comment by rautenrieth-da** [comment]
Given the points above, I don't think not validating the parties in the ledger API server can be exploited for anything. The validation would however be helpful for detecting problems in the ledger setup. If we do decide to validate the parties, we should:

1. Use a cache to prevent adding database queries
2. Validate all requests that involve a party field (almost all non-admin services), not just some of them
